relay-kafka/src/producer/mod.rs

```diff
@@ -139,6 +140,9 @@ impl fmt::Debug for Producer {
 #[derive(Debug)]
 pub struct KafkaClient {
     producers: HashMap<KafkaTopic, Producer>,
+    // rate limits are stored separately from producers because producers may be reused across
+    // topics if they have the same broker config. we shouldn't do that for rate limit state!
+    rate_limiters: HashMap<KafkaTopic, KafkaRateLimits>,
```

Comment:
Is it annoying to collapse these hashmaps into a single one? Feels appropriate to have a mapping from `topic -> state`

Alternatively, I think it makes sense to just move the rate limiter into the producer, that also seems like the better approach when you have to modify the headers.

Reply:
i realized i misunderstood how `Producer` gets deduped. only the rdkafka producer is arced. was able to move everything into Producer.